sequencing-server/src/lib/codegen/CommandEDSLPreface.ts

```diff
@@ -483,30 +483,30 @@ function commandsWithTimeValue<T extends TimingTypes>(
   }, {} as typeof Commands);
 }
 
-// @ts-ignore
-function orderCommandArguments(args: { [argName: string]: any }, order: string[]): any {
-  return order.map(key => args[key]);
-}
+function flatten(input: { [argName: string]: any }): any {
+  let flatList: any[] = [];
```

Comment:
Similarly, arrays can be allocated with `const` and pushed to - the array reference is const, but not the underlying contents. It's a bit of a sore point of TypeScript that has caused much conversation about the "scope" of the `const` implications and how it might differ from JavaScript behavior.

Reply:
The problem is we need to look at the first element in order to see if the list is already flattened, I can add a check like:
```
const elements = Object.values(input);

if (elements && elements.length > 0 && typeof elements[0] !== 'object') {
```

But I figured that was less readable than just allocating `flatList` first.

Reply:
I would prefer readability since the allocating is so minor.